source/shared/cpp/ObjectModel/MarkDownParser.cpp

```diff
@@ -0,0 +1,425 @@
+                              if (!isFound)
+                              {
+                                  // put back everything
+                                  while (!isFound && !store.empty())
+                                  {
+                                      stk.push_back(store.back());
+                                      store.pop_back();
+                                  }
+
+                                  //right emphasis becomes left emphasis
+                                  auto elem_to_be_inserted = curr_left_delim;
+                                  elem_to_be_inserted++;
+                                  m_leftLookUpTable.insert(elem_to_be_inserted, *top_right);
+                                  curr_left_delim = --elem_to_be_inserted;
+                                  stk.push_back(curr_left_delim);
+
+                                  auto elem_to_erase = top_right;
+                                  // move to next token for right delim tokens
+                                  top_right++;
+                                  m_rightLookUpTable.erase (elem_to_erase);
+                                  // no maching found begin from the start
+                                  continue;
+                              }
+                          }
+                      }
+                }
+                // check which one will have leftover delims
+                leftOver = curr_left_delim->m_emphCnts - top_right->m_emphCnts;
+                delimCnts = AdjustEmphasisCounts(leftOver, curr_left_delim, top_right);
+                right_token_idx = top_right;
+                if (leftOver >= 0)
+                {
+                    lookBehind ^= curr_left_delim_type;
+                }
+            }
+            else
+            {
+                // Rule #14, when two overraps, prefer left side of the overrap
+                if (stk.size () > 1 && lookBehind & curr_right_delim_type) 
+                {
+                    char ch = (curr_right_delim_type == Asterisk)? '*' :'_';
+                    // pop until matching delim is found
+                    while (!stk.empty() && m_tokenizedString[curr_left_delim->m_idx][0] != ch)
+                    {
+                        stk.pop_back();
+                        curr_left_delim = stk.back();
+                    }
+
+                    leftOver = curr_left_delim->m_emphCnts - top_right->m_emphCnts;
+                    delimCnts = AdjustEmphasisCounts(leftOver, curr_left_delim, top_right);
+                    right_token_idx = top_right;
+                    lookBehind = 0;
+                }
+            }
+
+            // emphasis found
+            if (delimCnts % 2)
+            {
+                curr_left_delim->m_tags.push_back(EmphasisItalic);
+                top_right->m_tags.push_back(EmphasisItalic);
+            }
+
+            // strong emphasis found
+            for (int rpts = 0; rpts < delimCnts / 2; rpts++)
```

Comment:
>rpts [](start = 21, length = 4)

rpts is repetitions?  Either give this a more descriptive name, or, alternatively, in the case of a for loop like this where the loop variable itself isn't too important it's ok to just call it "i"